### DISCUSSION TITLE
3D localization of neurons in HNN

### DISCUSSION THREAD

**Discussion by ntolley** [jonescompneurolab/hnn-core#657] [neuro]
I've been working on modifying how neurons in the model are positioned in 3D space and running into some points that could be improved or clarified:

- The default cell separation is currently way too small (1 micron), perhaps net.set_cell_positions(inplane_distance=10) is more appropriate?
- What is the rationale of having `space_constant=3.0` in the evoked drive? Is this 
- Related to the above, it seems like there should be an option to simply set a fixed weight with no spatial dependency. The best we can do right now is to set the lamtha/space constant to a really huge number
- Since the evoked drive weights vary with space, the location of the artificial cell matters quite a lot when you start changing the network.

Also we need to set some sort of tolerance/threshold on the distance dependent `delay`. The delay is calculated as follows:
```py
delay = nc_dict['A_delay'] / _calculate_gaussian(cell_dist, 1, scaled_lamtha)
```

When the cell dist is large, the gaussian evaluates to approximately zero, which produces an overflow when we calculate its reciprocal. Maybe the solution is to set the delay to `nan` or something and just remove the offending connections?

**Comment by jasmainak** [comment]
I propose that `net.cell_cell_positions` should have an additional parameter `scale_lamtha` which is `True` by default. It will go over all the connections in the network and scale the lamtha by the same ratio as the inplane distance. Wdyt?

At the moment, I am not sure what the units of `lamtha` are? (we claim to work with physical units)

I would advocate against hackish solutions like using `nan` when the computation overflows

**Reply by rythorpe** [reply]
My understanding is that `lamtha` currently represents the connection weight between neurons based on their distance in indices (e.g., `lamtha = 1.0` means that the synaptic weight falls off to 1/e at one grid position away) and that `inplane_distance` only applies to extracellular recordings.

**Comment by jasmainak** [comment]
We could have an option `lamtha=None` which makes the weights uniform ...

**Reply by rythorpe** [reply]
+1
